packages/clarity-js/src/data/metadata.ts

```diff
@@ -64,13 +64,39 @@ export function start(): void {
   track(u);
 }
 
+export function userAgentData(): void {
+  if (navigator["userAgentData"] && navigator["userAgentData"].getHighEntropyValues) {
+    navigator["userAgentData"].getHighEntropyValues(
+      ["model",
+      "platform",
+      "platformVersion",
+      "uaFullVersion"])
+      .then(ua => { 
+        dimension.log(Dimension.Platform, ua.platform); 
+        dimension.log(Dimension.PlatformVersion, ua.platformVersion); 
+        ua.brands.forEach(brand => {
+          dimension.log(Dimension.Brand, brand.name + Constant.Tilde + brand.version); 
+        });
+        dimension.log(Dimension.Model, ua.model); 
+        metric.max(Metric.Mobile, ua.mobile ? BooleanFlag.True : BooleanFlag.False); 
+      });
+  }
+}
+
 export function stop(): void {
   callback = null;
   rootDomain = null;
+  data = null;
 }
 
-export function metadata(cb: MetadataCallback): void {
-  callback = cb;
+export function metadata(cb: MetadataCallback, onupgrade: boolean = true): void {
+  if (data && onupgrade === false) {
+    // Immediately invoke the callback if the caller explicitly doesn't want to wait for the upgrade confirmation
+    cb(data, !config.lean);
```

Comment:
```suggestion
    cb(data, onupgrade);
```

Reply:
nit: we should also change https://github.com/microsoft/clarity/blob/ddac23ec718f702e9761642a1fdf3b094ff6eb6c/packages/clarity-js/src/data/metadata.ts#L132 to be `callback(data, true)` for readability

Reply:
This won't be correct since we want to indicate if a valid playback is available for this particular page or not.